holoviews/core/data/pandas.py

```diff
@@ -346,20 +382,24 @@ def select(cls, dataset, selection_mask=None, **selection):
 
     @classmethod
     def values(
-            cls,
-            dataset,
-            dim,
-            expanded=True,
-            flat=True,
-            compute=True,
-            keep_index=False,
+        cls,
+        dataset,
+        dim,
+        expanded=True,
+        flat=True,
+        compute=True,
+        keep_index=False,
     ):
         dim = dataset.get_dimension(dim, strict=True)
-        data = dataset.data[dim.name]
+        isindex = cls.isindex(dataset, dim)
+        if isindex:
+            data = cls.index_values(dataset, dim)
```

Comment:
I genuinely don't know but does `index.get_level_values(dimension.name)` always return an array of the same length as the DataFrame itself?

Reply:
I think so, the way thinks are structured behind the scenes looks like a list of tuples for MultiIndex.

![image](https://github.com/holoviz/holoviews/assets/19758978/b4717c31-ef3b-4e4d-b875-2aead6f314f7)

``` python
import pandas as pd

frame = pd.DataFrame({"number": [1, 1, 2, 2], "color": ["red", "blue", "red", "blue"]})
index = pd.MultiIndex.from_frame(frame, names=frame.columns)
df = pd.DataFrame(range(4), index=index, columns=["values"])
df.index.get_level_values("number")